tests/raftstore/test_transport.rs

```diff
@@ -0,0 +1,107 @@
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+use std::collections::HashSet;
+
+fn split_leader_with_majority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    let mut s1 = HashSet::new();
```

Comment:
I'd also come up with that idea at the first sight, later I found the code more ugly, something like:

    if leader located first half {
        for ... {
            copy to s1
        }
        for ... {
            copy to s2
        }
    } else {
        for ... {
        
        }
        for ... {
        
        }
    }

have to take more think about +-1 position is also ugly